### DISCUSSION TITLE
Wpf net8 browser and request listeners

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [cefsharp/CefSharp#4856] [General]
Hello, I am getting started with [_Wpf and net8_](4787) and wanting a good embedded Chromium browser, came across this one being a strong, vibrant project. I do need to facilitate automating bits of certain pages at runtime, if possible. Also, moreover, and more importantly, if possible, I need to also listen to outgoing requests. Is something like this possible with Cef in general, or CefSharp? Cheers thanks... 🍻

**Comment by amaitland** [comment]
Hi,

If your new to the project then https://github.com/cefsharp/CefSharp/wiki/General-Usage is a great place to start.

https://github.com/cefsharp/CefSharp/wiki/General-Usage#request-handling should give you some hints on request handling.

If you can provide more details I can probably point you in the direction of a better sample.

**Reply by mwpowellhtx** [reply]
Sure, great, I will look into those couple of links. Mainly, I want to watch request Urls, whether they were the browser address or not, and parse out some of the arguments for internal use; i.e. probably some page element AJAX REST request going on.

**Reply by mwpowellhtx** [reply]
Just reviewing some of the getting started notes, dispatch timers, definitely something I'd like to comprehend... but in what context, I am not hundred percent clear from the [_message loop_](https://github.com/cefsharp/CefSharp/wiki/General-Usage#multithreadedmessageloop) guidance. Is that within the scope of a `Window`?

But beyond that, is there a top level _"browser"_ component that I drop into said window? Or rather even _that_ is the base class analog to `Window`?

**Reply by mwpowellhtx** [reply]
I am mostly there, at least out of the gate and walking... Arranged window with the browser re: the WPF minimal example. Pretty helpful in getting started.

Next up is to think about how we intercept requests, but before that, working with the DOM if possible. i.e. to inject things like a user login, knowing what those form controls are, and injecting from my WPF widgets.

**Reply by amaitland** [reply]
> Just reviewing some of the getting started notes, dispatch timers, definitely something I'd like to comprehend... but in what context, I am not hundred percent clear from the [_message loop_](https://github.com/cefsharp/CefSharp/wiki/General-Usage#multithreadedmessageloop) guidance. Is that within the scope of a `Window`?

What's your question exactly?

**Comment by mwpowellhtx** [comment]
Next bit of what I may need is DOM, how to work with it, etc. I have a sense where to find that, but any guidance is also appreciated. Cheers 🍻

**Reply by mwpowellhtx** [reply]
Wow getting started with [`CefSharp.Dom`](https://nuget.org/packages/CefSharp.Dom), so many direct and indirect extension dependencies, logging, primitives, etc. Lagging a bit for [`Newtonsoft.Json`](https://www.nuget.org/packages/Newtonsoft.Json/) however `13.0.3` at the time of this entry. I'm surprised there are not any security warnings being raised. Would we be save to upgrade that? If for nothing else, we will probably be using that dependency as well, and do not want to be shackled to `10`.

**Reply by mwpowellhtx** [reply]
[_Semi DOM related question_](https://github.com/cefsharp/CefSharp.Dom/discussions/53), but also rooted, I think, in how to host and operate a browser window in a kosher, i.e. non-disposing, manner.

**Reply by mwpowellhtx** [reply]
[_Some new observations_](https://github.com/cefsharp/CefSharp/discussions/4031#discussioncomment-9962948).

**Reply by mwpowellhtx** [reply]
Victory! See steps within observations. 🎆 🔥

**Reply by amaitland** [reply]
> Lagging a bit for [`Newtonsoft.Json`](https://www.nuget.org/packages/Newtonsoft.Json/) however `13.0.3` at the time of this entry. I'm surprised there are not any security warnings being raised. Would we be save to upgrade that? If for nothing else, we will probably be using that dependency as well, and do not want to be shackled to `10`.

It's only the minimum version being `10`, you should be fine to use `13`.
